python/tank/bootstrap/configuration_writer.py

```diff
@@ -28,6 +28,50 @@
 log = LogManager.get_logger(__name__)
 
 
+class RenameGuard(object):
```

Comment:
wondering if this could be useful outside of this context. perhaps this should live in `tank/util/filesystem.py`?